### PR TITLE
fn for determining zodiac sign

### PR DESCRIPTION
**Pull request by gsbelarus**

### CODE REVIEW

File: src/zodiac.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    return 'Libra';
+  } else if ((month === 10 && day >= 23) || (month === 11 && day <= 21)) {
+    return 'Scorpio';
+  } else if ((month === 11 && day >= 22) || (month === 12 && day <= 21)) {
+    return 'Sagittarius';
+  } else {
+    throw new Error('Invalid date');
+  }
+}
\ No newline at end of file
```
Comment (gsbelarus) [inline src/zodiac.ts:31]: Here is a brief review of the provided code patch:

1. **Type Definition**: It's good to see that you're using a TypeScript union type for `ZodiacSign`, which ensures that the function can only return valid zodiac signs.

2. **Function Parameters**: The parameters are correctly typed as numbers, which is expected for days and months.

3. **Algorithm Logic**:
   - The logic seems accurate regarding the dates for each zodiac sign.
   - Aquarius is missing from the conditionals, which is definitely an oversight. There should be conditions to check for Aquarius just like the other signs.
   - One notable bug due to the missing Aquarius condition is that if someone has a birthday between January 21 and February 18, it will throw an error instead of returning 'Aquarius'.
  
4. **Error Handling**: 
   - The `throw new Error('Invalid date')` at the end will handle any day/month combination outside of the conditions specified, which is good. However, it might be too generic as it could also catch valid dates (like those in January and February because 'Aquarius' is not represented in the code).
   - The error message could also include the invalid input (month and day) to aid debugging.

5. **Input Validation**: 
   - Before comparing dates, consider adding validation for the day and month values. For instance, ensure that the month is between 1-12 and the day is within valid ranges for each month (taking into account leap years).
   - Additionally, the function does not handle cases where the `day` is less than 1 or greater than the maximum days in the specified `month`.

6. **Performance**: 
   - The if-else pattern used here is the straightforward approach and is fine for the number of conditions involved.

7. **Readability**: 
   - The function is relatively easy to understand. 

8. **Code Formatting**: 
   - It is generally recommended to have a newline at the end of a file, which is missing here (\ No newline at end of file).

9. **Testing**: 
   - Make sure there are corresponding unit tests that cover all edge cases, such as the boundaries of each zodiac sign, invalid dates, as well as leap year considerations.

**Suggested improvements**:
- Add handling for the Aquarius zodiac sign.
- Improve input validation to check for correct `day` ranges depending on the month, including leap years.
- Enhance the error message by including what the invalid inputs were.
- Consider adding a newline at the end of the file.

In summary, the main functional issue with the given code is the absence of the 'Aquarius' sign condition. Input validation should be improved, and providing more descriptive error messages would enhance debuggability.
